src/strategies/stratCraftItem.ts

```diff
@@ -28,6 +29,27 @@ export class StratCraftItem extends StrategyBase
 
 class CraftItemInstance extends StrategyExecutionInstance
 {
+    canCraft(recipe: Recipe): boolean
+    {
+        // TODO check materials in inventory
```

Comment:
something like https://github.com/rom1504/rbot/blob/master/task.js#L395 can help here

Reply:
Thanks for the link!